runtime/middleware/middleware.go

```diff
@@ -0,0 +1,102 @@
+// Package middleware provides middleware functionality for defining
+// generic processing across multiple API endpoints, typically for
+// cross-cutting concerns like validation, caching, or error monitoring.
```

Comment:
Can you add a link to the middleware docs page? We've done this on each of the other package docs.